OpenSkyPlusApi/OpenSkyPlusApi.cs

```diff
@@ -1,722 +1,754 @@
+        sb.AppendLine($"\tBackspin: {shot.Spin.Backspin}");
+        sb.AppendLine($"\tSide Spin: {shot.Spin.SideSpin}");
+        sb.AppendLine($"\tTotal Spin spin: {shot.Spin.TotalSpin}");
+        sb.AppendLine($"\tMeasurement Confidence: {shot.Spin.MeasurementConfidence}");
+        _logger.LogInfo(sb.ToString());
+    }
+
+    public void LogToOpenSkyPlus(string message, object level)
+    {
+        try
+        {
+            switch ((LogLevels)level)
+            {
+                case LogLevels.Info:
+                    _logger.LogInfo(message);
+                    break;
+                case LogLevels.Warning:
+                    _logger.LogWarning(message);
+                    break;
+                case LogLevels.Debug:
+                    _logger.LogDebug(message);
+                    break;
+                case LogLevels.Error:
+                default:
+                    _logger.LogError(message);
+                    break;
+            }
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Failed to translate logging request to {_logger.GetType().Name}: {ex}");
+        }
+    }
+
+    /// <summary>
+    ///     Logs all the raw data from a shot.
+    ///     Useful for understanding what we are looking at. We probably don't need it anymore,
+    ///     but it was annoying to write, so I'm leaving it here in case its needed later.
+    /// </summary>
+    private void ShotDump(object shot)
+    {
+        var Shot = shot.GetType();
+
+        var FDDDFMHPPEP = Shot.GetField("FDDDFMHPPEP",
+                BindingFlags.Public | BindingFlags.Instance)
+            .GetValue(shot);
+
+        var GGEGJMKCNOP = Shot.GetField("GGEGJMKCNOP",
+                BindingFlags.Public | BindingFlags.Instance)
+            .GetValue(shot);
+
+        var MHLDAHLDJFE = Shot.GetField("MHLDAHLDJFE",
+                BindingFlags.Public | BindingFlags.Instance)
+            .GetValue(shot);
+
+        var LDLMHJGLBDF = Shot.GetField("LDLMHJGLBDF",
+                BindingFlags.Public | BindingFlags.Instance)
+            .GetValue(shot);
+
+        var DLIIJJNCIPP = Shot.GetField("DLIIJJNCIPP",
+                BindingFlags.Public | BindingFlags.Instance)
+            .GetValue(shot);
+
+        /////////////////
+        // FDDDFMHPPEP //
+        /////////////////
+
+        var FDDDFMHPPEP_T = FDDDFMHPPEP.GetType();
+
+        _logger.LogDebug("FDDDFMHPPEP:");
+
+        var BAFOBJEAGMN = (float)FDDDFMHPPEP_T.GetField("BAFOBJEAGMN",
+                BindingFlags.Public | BindingFlags.Instance)
+            .GetValue(FDDDFMHPPEP);
+        _logger.LogDebug($"BAFOBJEAGMN: {BAFOBJEAGMN}");
+
+        var GMBPBDLKJEG = (float)FDDDFMHPPEP_T.GetField("GMBPBDLKJEG",
+                BindingFlags.Public | BindingFlags.Instance)
+            .GetValue(FDDDFMHPPEP);
+        _logger.LogDebug($"GMBPBDLKJEG: {GMBPBDLKJEG}");
+
+        var EDNNKEPLHKC = (float)FDDDFMHPPEP_T.GetField("EDNNKEPLHKC",
```

Comment:
Strong typing is preferred so they fail should we end up pulling the wrong value or something